src/Webnative/Path/Encapsulated.elm

```diff
@@ -0,0 +1,46 @@
+module Webnative.Path.Encapsulated exposing (toDirectory, toFile)
+
+{-|
+
+
+# Encapsulated Paths
+
+@docs toDirectory, toFile
+
+-}
+
+import Webnative.Path as Path exposing (Directory, Encapsulated, File, Kind(..), Path, directory, file)
+
+
+
+-- EXTRACTION
+
+
+{-| Remove the membrane and extract a `Path Directory`.
+-}
+toDirectory : Path Encapsulated -> Result String (Path Directory)
+toDirectory path =
+    case Path.kind path of
+        Directory ->
+            path
+                |> Path.unwrap
+                |> directory
+                |> Ok
+
+        File ->
+            Err "Encapsulated path is a File path, expected Directory"
```

Comment:
I don't think the error message adds much here.
The contrary, the Result type might indicate that there's multiple different errors that could happen.
Also, I worry that the strings will just get surfaced to a UI at some point. Instead, apps should provide their own, custom error message.
What if we just went `Path Encapsulated -> Maybe (Path Directory)`?

Reply:
Sounds good 👍